### PR TITLE
fix(config): Remove the deprecated configs.

### PR DESCRIPTION
**Pull request by jallamsetty1**
Add the new codec selection settings for desktop and mobile endpoints.

### CODE REVIEW

File: react/features/base/config/reducer.ts [hunk 1]
```diff
@@ -47,13 +47,11 @@ const INITIAL_RN_STATE: IConfig = {
...
     p2p: {
-        disabledCodec: 'vp9',
-        preferredCodec: 'h264'
+        mobileCodecPreferenceOrder: [ 'H264', 'VP8' ]
     },
 
     videoQuality: {
-        disabledCodec: 'vp9',
-        preferredCodec: 'vp8'
+        mobileCodecPreferenceOrder: [ 'VP8', 'VP9' ]
```
Comment (saghul) [inline react/features/base/config/reducer.ts:54]: Wouldn't this be just "VP8" ?

File: react/features/base/config/configWhitelist.ts [hunk 2]
```diff
@@ -195,7 +195,6 @@ export default [
     'p2p',
     'participantsPane',
     'pcStatsInterval',
-    'preferredCodec',
```
Comment (saghul) [inline react/features/base/config/configWhitelist.ts:198]: We need to keep backwards compat, so this still needs to be whitelisted.

File: config.js [hunk 3]
```diff
@@ -427,25 +427,12 @@ var config = {
 
     // Specify the settings for video quality optimizations on the client.
     // videoQuality: {
-    //    // Provides a way to prevent a video codec from being negotiated on the JVB connection. The codec specified
```
Comment (saghul) [inline config.js:430]: Please put the dreprecated ones at the end of video quality with a DEPRECATED title, see how we've done that with other options.

File: react/features/base/config/reducer.ts [hunk 4]
```diff
@@ -47,13 +47,11 @@ const INITIAL_RN_STATE: IConfig = {
     // FIXME: than requiring this override here...
 
     p2p: {
-        disabledCodec: 'vp9',
-        preferredCodec: 'h264'
+        mobileCodecPreferenceOrder: [ 'H264', 'VP8' ]
```
Comment (saghul) [inline react/features/base/config/reducer.ts:50]: What the value if we don't define this? We could make the code behave like so, and remove this override entirely, and people can configure whatever they want in config.js, no?

Comment (jallamsetty1) [reply react/features/base/config/reducer.ts:50]: https://github.com/jitsi/lib-jitsi-meet/pull/2296/files#diff-dcb81f0be0b559eee3d7693e162f67f90eee41d6fa0a8cbc91c6460ec02f5c36R38 will be the new default order which is applicable to desktop clients. We could define a different set for mobile endpoints by default.

File: react/features/base/config/configWhitelist.ts (hunk 2 above)
Comment (jallamsetty1) [reply react/features/base/config/configWhitelist.ts:198]: The `videoQuality` and `p2p` sections are whitelisted so we don't need `preferredCodec` to be whitelisted for the deprecated settings to work. I think it was added by mistake.

File: react/features/base/config/reducer.ts (hunk 4 above)
Comment (saghul) [reply react/features/base/config/reducer.ts:50]: Let's do that please!

**Comment by saghul** [comment]
Other PRs seem to pass. Are we sure the failures are not related?

**Comment by jallamsetty1** [comment]
MuteTest on Firefox has been failing on alpha for few weeks now so that one is unrelated for sure.

**Review by saghul — APPROVED** [review]

**Pull request merged**